site/gatsby-site/server/fields/common.ts

```diff
@@ -90,140 +88,82 @@ export interface UserAdminData {
 
-    for (const userData of response) {
+    if (!user) {
 
-        if (userData.data?.email) {
+        const userApiResponse = await apiRequest({ path: `/users/${userId}` });
 
-            const user: UserAdminData = {};
+        if (userApiResponse.data) {
 
-            user.email = userData.data.email;
-            user.creationDate = new Date(userData.creation_date * 1000);
-            user.lastAuthenticationDate = new Date(userData.last_authentication_date * 1000);
-            user.disabled = userData.disabled;
-            user.userId = userData._id;
+            user = {
+                email: userApiResponse.data.email,
+                creationDate: new Date(userApiResponse.creation_date * 1000),
+                lastAuthenticationDate: new Date(userApiResponse.last_authentication_date * 1000),
+                disabled: userApiResponse.disabled,
+                userId,
+            }
 
-            result.push(user);
+            usersCache.push(user);
         }
     }
 
-    return result;
-}
-
-interface SendEmailParams {
-    recipients: {
-        email: string;
-        userId: string;
-    }[];
-    subject: string;
-    dynamicData: {
-        incidentId?: string;
-        incidentTitle?: string;
-        incidentUrl?: string;
-        incidentDescription?: string;
-        incidentDate?: string;
-        developers?: string; // HTML string of developers
-        deployers?: string;   // HTML string of deployers
-        entitiesHarmed?: string; // HTML string of harmed entities
-        reportUrl?: string;   // URL for a specific report (optional)
-        reportTitle?: string; // Title of the report (optional)
-        reportAuthor?: string; // Author of the report (optional)
-        entityName?: string;   // Entity name (optional)
-        entityUrl?: string;    // Entity URL (optional)
-    };
-    templateId: string; // Email template ID
-}
-
-export const replacePlaceholdersWithAllowedKeys = (template: string, data: { [key: string]: string }, allowedKeys: string[]): string => {
-    return template.replace(/\{\{\s*(\w+)\s*\}\}/g, (match, key) => {
-        return allowedKeys.includes(key) && key in data ? data[key] : match;
-    });
+    return user;
 }
 
-export const mailersendBulkSend = async (emails: EmailParams[]) => {
-
-    const mailersend = new MailerSend({
-        apiKey: config.MAILERSEND_API_KEY,
-    });
-
-    await mailersend.email.sendBulk(emails);
-}
 
-export const sendEmail = async ({ recipients, subject, dynamicData, templateId }: SendEmailParams) => {
+let cachedToken: string | null = null;
+let tokenExpiration: number | null = null;
 
-    const emailTemplateBody = templates[templateId];
+/**
+ * Fetches the access token for the MongoDB Atlas Admin API.
+ * 
+ * Note: The token is cached to speed up subsequent requests. Tokens expire after 30 minutes.
+ * 
+ * @returns {Promise<string>} A promise that resolves to the access token.
+ */
+const getAccessToken = async () => {
```

Comment:
This is not specifically related to the fix.

But it will help speed up the notification process and every other request.